apps/web/src/views/BuyCrypto/containers/CryptoQuoteForm.tsx

```diff
@@ -1,27 +1,23 @@
 import { Dispatch, SetStateAction, useEffect, useState, useRef } from 'react'
-import { BuyCryptoState } from 'state/buyCrypto/reducer'
 import { useTranslation } from '@pancakeswap/localization'
 import { useActiveChainId } from 'hooks/useActiveChainId'
-import { CryptoFormView } from 'views/BuyCrypto/types'
+import { CryptoFormView, ProviderQoute } from 'views/BuyCrypto/types'
 import { FormHeader } from './FormHeader'
-import { ProviderQoute } from '../hooks/usePriceQuoter'
-import { FormContainer } from '../components/FormContainer'
+import { FormContainer } from './FormContainer'
 import Accordion from '../components/AccordianDropdown/Accordian'
 
 export function CryptoQuoteForm({
   setModalView,
-  buyCryptoState,
-  combinedQuotes,
   fetchQuotes,
+  combinedQuotes,
 }: {
   setModalView: Dispatch<SetStateAction<CryptoFormView>>
-  buyCryptoState: BuyCryptoState
-  combinedQuotes: ProviderQoute[]
   fetchQuotes: () => Promise<void>
+  combinedQuotes: ProviderQoute[]
 }) {
   const { t } = useTranslation()
   const { chainId } = useActiveChainId()
-  const [timer, setTimer] = useState(0)
+  const [timer, setTimer] = useState(30)
```

Comment:
Let's use `react-query`. You can specify the refetch interval

Reply:
hmm its tricky to integrate react query here because fetchQuotes returns void. i just use it as a callback to update the quotes state. also i need a timer state here so we can display to the ui the current second left until refetch. so using react query here with a refetch interval might be slighly decoupled from the timer state var? what do you think

Reply:
You can use react-query with any async functions. Also you don't need to handle the logic in the effect hook anymore?
```typescript
const { isFetching } = useQuery('', fetchQuotes, {
  refetchInterval: () => 30 * 1000,
})
```

Reply:
ah sorry didn't see we need a timer to display here. Ok then can we have a hook called `useCountdown`? So we can make the code cleaner and reuse in the future. I'm thinking something like this
```typescript
const [secondsRemaining, restart] = useCountdown(30, { onEnd: fetchQuotes })
```

Reply:
It's more of a resusable hook and can move the logic out of the form component

Reply:
resolved, including in pr for add wbtc as a new currency